lib/rag/index.ts

```diff
@@ -178,7 +208,7 @@ export class LisaRagStack extends Stack {
                         version: EngineVersion.OPENSEARCH_2_9,
                         enableVersionUpgrade: true,
                         vpc: vpc.vpc,
-                        vpcSubnets: vpc.subnetSelection ? [vpc.subnetSelection] : [],
+                        ...vpc.subnetSelection ? {vpcSubnets: [vpc.subnetSelection]} : {},
```

Comment:
Can be simplified to: 
`
...(vpc.subnetSelection && {vpcSubnets: [vpc.subnetSelection]})
`